### PR TITLE
`Programming exercises`: Improve button labels for Orion integration

### PR DESCRIPTION
**Pull request by Kroko-fant**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- No server changes made
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context

The PR improves the labelling on buttons the integration between Orion and Artemis more
### Steps for Testing

Artemis:
1. No extra buttons should appear on pages related to Orion exercises

With Orion:
1. No Button should have a IntelliJ logo in the button
2. Local testing should work with the newest Orion version and Artemis should provide the right Test Repository URL and initialize the test repo.

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [orion-exercise-assessment-dashboard.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5863/latest/artifact/shared/Coverage-Report-Client-Tests/app/orion/assessment/orion-exercise-assessment-dashboard.component.ts.html) | 100% | ✅ |
| [orion-programming-exercise.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5863/latest/artifact/shared/Coverage-Report-Client-Tests/app/orion/management/orion-programming-exercise.component.ts.html) | 100% | ✅ |
| [orion-exercise-details-student-actions.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5863/latest/artifact/shared/Coverage-Report-Client-Tests/app/orion/participation/orion-exercise-details-student-actions.component.ts.html) | 85.71% | ✅ ❌ |
| [orion-button.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5863/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/orion/orion-button/orion-button.component.ts.html) | 100% | ✅ |
| [orion-connector.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5863/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/orion/orion-connector.service.ts.html) | 93.75% | ✅ ❌ |
| [orion.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5863/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/orion/orion.ts.html) | 100% | ✅ |

## Summary by CodeRabbit

- **New Features**
  - Implemented an "Orion open button" for programming exercises, allowing users to open assessments directly in Orion from a tutoring view.
  - Enhanced button labels and functionality within the Orion integration to improve user experience and context awareness.
  - Introduced new Orion button types and updated icons to reflect various user actions like download, submit, and reload.

- **Enhancements**
  - Updated UI elements to shift context from IntelliJ to Orion for relevant user actions.
  - Added new methods to improve interaction with the Orion programming environment, such as initializing test repositories.

- **Bug Fixes**
  - Fixed conditional display logic for Orion-specific buttons and icons to ensure accurate representation of the current state.

### CODE REVIEW

**Review by lennart-keller — APPROVED** [review]
Tested in testing session. Works well :)

**Review by yassinsws — APPROVED** [review]
Tested in a testing session

File: src/main/webapp/app/shared/orion/orion-connector.service.ts [hunk 1]
```diff
@@ -280,4 +280,17 @@ export class OrionConnectorService {
...
+     *
+     * @param exercise the current programming exercise
+     */
+    initializeTestRepository(exercise: ProgrammingExercise) {
+        if (exercise.testRepositoryUrl !== undefined) {
+            theWindow().orionExerciseConnector.initializeTestRepository(exercise.testRepositoryUrl);
+        } else {
+            theWindow().orionExerciseConnector.initializeTestRepository('null');
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Kroko-fant) [reply src/main/webapp/app/shared/orion/orion-connector.service.ts:295]: since the function explicitly takes non null values this is intentional behavior

**Review by nityanandaz — APPROVED** [review]
Manual test: demonstrated on TS5

**Review by basak-akan — APPROVED** [review]
Demostrated during the testing session worked well, also code lgtm

**Review by nata-81 — APPROVED** [review]
Demonstrated during testing session works well.

**Review by Jan-Thurner — APPROVED** [review]
Demonstrated in a TS, works as described.

**Pull request merged**